### PR TITLE
[kotlin-client] Allowing vendor types for json

### PR DESCRIPTION
**Pull request by sheepdreamofandroids**
When (de)serializing json bodies, only the exact "application/json" mimetype was matched. Now vendor types like "application/x-mycompany.myapp-v2+json" will also be sent and recognized.

In addition, the "Content-Type" and "Accept" headers are set. Content-Type gets the first mimetype from the swagger file that can actually be serialized while Accept gets all deserializable mimetypes.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@jimschubert (2017/09) ❤️, @dr4ke616 (2018/08) @karismann (2019/03) @Zomzog (2019/04) @andrewemery (2019/10) @4brunu (2019/11) @yutaka0m (2020/03)

### CODE REVIEW

**Comment by sheepdreamofandroids** [comment]
I keep merging master into my branch but the bitrise task keeps failing. The same thing happens in other PR's as well so I'm pretty sure that it is not a problem with the code changes in this PR. It looks like the iOS machine has connection problems.

**Comment by 4brunu** [comment]
Hey @sheepdreamofandroids,
Thanks for opening this PR.
Did you test those changes in a project of your that use vendor type json extension?
Let's also way for @wing328 to take a look at this and see his opinion.
Thanks

**Comment by sheepdreamofandroids** [comment]
> Did you test those changes in a project of your that use vendor type json extension? 
Yes, I did. Worked as expected.

Note that this change also affects clients without any vendor types by sending Accept headers for only the types that can actually be deserialized.

**Comment by sheepdreamofandroids** [comment]
Could somebody take a look at this please?

File: samples/client/petstore/kotlin-gson/docs/PetApi.md [hunk 1]
```diff
@@ -155,7 +155,7 @@ Configure petstore_auth:
 ### HTTP request headers
 
  - **Content-Type**: Not defined
- - **Accept**: application/xml, application/json
```
Comment (4brunu) [inline samples/client/petstore/kotlin-gson/docs/PetApi.md:158]: @sheepdreamofandroids is this change expected?

Comment (sheepdreamofandroids) [reply samples/client/petstore/kotlin-gson/docs/PetApi.md:158]: Yes, this client doesn't support xml.

**Review by 4brunu — APPROVED** [review]
Looks good to me 👍
@wing328 this PR is ready to review and merge

**Pull request merged**